circadian/lights.py

```diff
@@ -4,157 +4,400 @@
+        if not isinstance(repeat_period, (float, int)):
+            raise ValueError(f"repeat_period must be a float or int, got {type(repeat_period)}")
+    if ax is not None:
+        if not isinstance(ax, plt.Axes):
+            raise ValueError(f"ax must be a matplotlib Axes object, got {type(ax)}")
+    if num_samples is not None:
+        if not isinstance(num_samples, int):
+            raise ValueError(f"num_samples must be an int, got {type(num_samples)}")
+    
+    if plot_start_time is None:
+        plot_start_time = self.start_time
+    if plot_end_time is None:
+        plot_end_time = self.end_time
         
-    @staticmethod
-    def ShiftWorkLight(lux: float = 150.0, # lux intensity of the light
+    t = np.linspace(plot_start_time, plot_end_time, num_samples)
+    vals = self.__call__(t, repeat_period=repeat_period)
+    if ax is None:
+        plt.figure()
+        ax = plt.gca()
+
+    ax.plot(t, vals, *args, **kwargs)
+    return ax
+
+# %% ../nbs/01_lights.ipynb 31
+@patch_to(Light)
+def RegularLight(lux: float=150.0, # lux intensity of the light
+                 lights_on: float=8.0, # hour of the day for lights to come on
+                 lights_off: float=16.0, # hour of the day for lights to go off
+                 ) -> 'Light':
+    "Create a light function that is on from `lights_on` to `lights_off` on a 24 hour schedule"
+    # type checking
+    if not isinstance(lux, (float, int)):
+        raise ValueError(f"lux must be a nonnegative float or int, got {type(lux)}")
+    elif lux < 0.0:
+        raise ValueError(f"lux must be a nonnegative float or int, got {lux}")
+    if not isinstance(lights_on, (float, int)):
+        raise ValueError(f"lights_on must be a float or int, got {type(lights_on)}")
+    if not isinstance(lights_off, (float, int)):
+        raise ValueError(f"lights_off must be a float or int, got {type(lights_off)}")
+
+    lights_on = np.mod(lights_on, 24.0)
+    lights_off = np.mod(lights_off, 24.0)
+    if lights_off > lights_on:
+        end_time = lights_on + (lights_off - lights_on) 
+        dark_section_before = Light(0.0, start_time=0.0, duration=lights_on)
+        light_section = Light(lux, 
+                        start_time=lights_on, 
+                        duration=lights_off - lights_on)
+        dark_section = Light(0.0, end_time, duration=24.0 - end_time)
+        return dark_section_before.concatenate(light_section).concatenate(dark_section)
+    else:
+        first_light = Light(lux, start_time = 0.0, duration=lights_off)
+        dark_section = Light(0.0, start_time=lights_off, duration=lights_on - lights_off)
+        second_light = Light(lux, start_time=lights_on, duration=24.0 - lights_on)
+        return  first_light.concatenate(dark_section).concatenate(second_light)
+
+# %% ../nbs/01_lights.ipynb 35
+@patch_to(Light)
+def ShiftWorkLight(lux: float = 150.0, # lux intensity of the light
                    dayson: int = 3, # number of days on the night shift
```

Comment:
let's use snake case for any variables that have multiple words; e.g. `days_on`

Reply:
Thanks for catching this!